### PR TITLE
Scrollpos methods

### PR DESCRIPTION
**Pull request by kmathew-pdftron**

### CODE REVIEW

File: android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java [hunk 1]
```diff
@@ -2459,6 +2465,26 @@ public static void onMethodCall(MethodCall call, MethodChannel.Result result, Vi
                 }
                 break;
             }
+            case FUNCTION_GET_SCROLL_POS: {
+                checkFunctionPrecondition(component);
+                try {
+                    getScrollPos(result,component);
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java:2471]: format

File: android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java [hunk 2]
```diff
@@ -3765,6 +3791,47 @@ private static void addBookmark(String title, Integer pageNumber, MethodChannel.
...
+    private static void getScrollPos(MethodChannel.Result result, ViewerComponent component) throws JSONException {
+        PDFViewCtrl pdfViewCtrl = component.getPdfViewCtrl();
+        JSONObject jsonObject = new JSONObject();
+
+        if (pdfViewCtrl == null) {
+            result.error("InvalidState", "PDFViewCtrl not found", null);
+            return;
+        }
+
+        jsonObject.put(REFLOW_ORIENTATION_HORIZONTAL,  pdfViewCtrl.getHScrollPos());
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java:3803]: format

File: android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java [hunk 3]
```diff
@@ -3765,6 +3791,47 @@ private static void addBookmark(String title, Integer pageNumber, MethodChannel.
...
+        jsonObject.put(REFLOW_ORIENTATION_VERTICAL, pdfViewCtrl.getVScrollPos());
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java:3804]: this key looks strange to me, as there is nothing to do with reflow here

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java [hunk 4]
```diff
@@ -3765,6 +3795,47 @@ private static void addBookmark(String title, Integer pageNumber, MethodChannel.
...
+    private static void setHorizontalScrollPosition(MethodCall call, MethodChannel.Result result, ViewerComponent component) {
+        PDFViewCtrl pdfViewCtrl = component.getPdfViewCtrl();
+        int horizontalScrollPosition = call.argument(KEY_HORIZONTAL_SCROLL_POSITION);
+        
+        if (pdfViewCtrl == null) {
+            result.error("InvalidState", "PDFViewCtrl not found", null);
+            return;
+        }
+
+        pdfViewCtrl.setHScrollPos((int) (horizontalScrollPosition + 0.5));
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java:3822]: unsure why the 0.5 is needed, original value is int, the method expects int as well

File: android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java [hunk 5]
```diff
@@ -3765,6 +3795,47 @@ private static void addBookmark(String title, Integer pageNumber, MethodChannel.
...
+    private static void setVerticalScrollPosition(MethodCall call, MethodChannel.Result result, ViewerComponent component) {
+        PDFViewCtrl pdfViewCtrl = component.getPdfViewCtrl();
+        int verticalScrollPosition = call.argument(KEY_VERTICAL_SCROLL_POSITION);
+
+        if (pdfViewCtrl == null) {
+            result.error("InvalidState", "PDFViewCtrl not found", null);
+            return;
+        }
+        
+        pdfViewCtrl.setVScrollPos((int) (verticalScrollPosition + 0.5));
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java:3835]: same here, why need to add 0.5

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

**Comment by sgong-pdftron** [comment]
too many conflicts

**Review by sgong-pdftron — APPROVED** [review]

**Review by dluco — APPROVED** [review]

**Pull request merged**